pcicrawler/ocp_diag.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+
+"""
+Copyright (c) Meta Platforms, Inc. and affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+import json
+import socket
+import pkg_resources
+import typing as ty
+import ocptv.output as tv
+from pci_lib import PCIDevice
+from abc import abstractmethod
+
+
+class OCPTestStep:
+    def __init__(self,
+                    device: PCIDevice,
+                    expected_result: str,
+                    ocp_run: tv.TestRun):
+        self._device = device
+        self._expected_result = expected_result
+        self._ocp_run = ocp_run
+        self._error_messages = []
+
+    @property
+    @abstractmethod
+    def _test_name(self):
+        pass
+
+    @abstractmethod
+    def run(self, current_value: str):
+        step = self._ocp_run.add_step(self._test_name)
+        if str(current_value) != str(self._expected_result):
+            step.add_diagnosis(tv.DiagnosisType.FAIL, 
+                            verdict="Test {} for device {}, expected {}, found {}.".format(
```

Comment:
you probably want `message` in all these Diagnoses, not `verdict`

> verdict: A short string describing the overall determination of the [diagnostic](https://github.com/opencomputeproject/ocp-diag-core/tree/main/json_spec#def-diagnostic) (e.g. cpu-good, hdd-overtemp, etc). These short human readable strings are typically used to aggregate counts of observed [symptoms](https://github.com/opencomputeproject/ocp-diag-core/tree/main/json_spec#def-symptom).

Reply:
mmm each of these tests are individual diagnostics in my head. so verdict seems to make sense?

Reply:
+1 to this mimir-d's comments here.

The verdict is intended to provide something that can be pareto'd up for analysis in aggregate rather than a more verbose message.

What you have in the verdict field here would be appropriate for the message field of the diagnosis.

In addition, the device{} seems to indicate a location or hwInfo where as the expected{} might also be a Measurement property.  I don't feel two strongly about the last two, but the verdict{} is definitely important.

We try not to include location or device identifiers in a verdict.